sympy/physics/mechanics/kane.py

```diff
@@ -317,6 +317,18 @@ def _initialize_constraint_matrices(self, config, vel, acc, linear_solver='LU'):
                 self._f_dnh = msubs(acc, udot_zero)
                 self._k_dnh = (acc - self._f_dnh).jacobian(self._udot)
 
+            # The linear velocity constraints must be linear in u so check
+            # for u in the components.
+            dy_syms = find_dynamicsymbols(self._f_nh.row_join(self._k_nh))
+            nonlin_vars = [vari for vari in self.u[:] + self._qdot[:]
```

Comment:
You removed the wording in the comment, but this still checks for qdots.

Reply:
> You removed the wording in the comment, but this still checks for qdots.

I will remove it there, too.